src/pages/game.js

```diff
@@ -33,141 +31,181 @@ export default function Game({data, location}) {
-      const isDraw = usersTurnCard[attribute] === computersTurnCard[attribute]
+      const hasUserWon = allState.usersCards[0][attribute] > allState.computersCards[0][attribute]
+      const isDraw = allState.usersCards[0][attribute] === allState.computersCards[0][attribute]
 
       if (isDraw) {
-         console.log(
-            `Draw: Users= ${attribute} - ${usersTurnCard[attribute]} \n Computers= ${attribute} - ${computersTurnCard[attribute]}`
-         )
-         setUsersCards([...usersCards, usersTurnCard])
-         setComputersCards([...computersCards, computersTurnCard])
-         setIsUsersTurn(!isUsersTurn)
-         setRoundWinner(false)
+
+         setAllState({
+            ...allState,
+            roundWinner: allState.currentPlayer === PLAYER_USER ? DRAW_PLAYER : DRAW_COMPUTER,
+            gameState: GAME_STATE_RESULTS,
+            selectedAttribute: attribute
+         })
          return
       }
 
       if (hasUserWon) {
-         console.log(
-            `User Won: Users= ${attribute} - ${usersTurnCard[attribute]} \n Computers= ${attribute} - ${computersTurnCard[attribute]}`
-         )
-         setUsersCards([...usersCards, usersTurnCard, computersTurnCard])
-         setIsUsersTurn(true)
-         setRoundWinner(PLAYER_USER)
-         setRoundsWon(roundsWon + 1)
+
+         setAllState({
+            ...allState,
+            roundsWon: allState.roundsWon + 1,
+            currentPlayer: PLAYER_USER,
+            roundWinner: PLAYER_USER,
+            gameState: GAME_STATE_RESULTS,
+            selectedAttribute: attribute
+         })
+
          return
       }
-      console.log(
-         `Computer Won: Users= ${attribute} - ${usersTurnCard[attribute]} \n Computers= ${attribute} - ${computersTurnCard[attribute]}`
-      )
-      setComputersCards([...computersCards, computersTurnCard, usersTurnCard])
-      setIsUsersTurn(false)
-      setRoundWinner(PLAYER_COMPUTER)
+
+      setAllState({
+         ...allState,
+         currentPlayer: PLAYER_COMPUTER,
+         roundWinner: PLAYER_COMPUTER,
+         gameState: GAME_STATE_RESULTS,
+         selectedAttribute: attribute
+      })
 
       return
    }
 
    useEffect(() => {
-      if (!turnCount) {
-         return
-      }
 
-      if (!usersCards.length || !computersCards.length) {
-         setGameWinner(usersCards.length ? PLAYER_USER : PLAYER_COMPUTER)
+      if (!allState.usersCards.length || !allState.computersCards.length) {
+         //Show results
          return
       }
 
-      setRoundWinner(null)
-      drawCard()
-      if (!isUsersTurn) {
-         computersTurn()
+      const cards = drawCard()
+
+      if (allState.roundWinner === null) {
+         return usersTurn(cards);
       }
-   }, [turnCount])
 
-   useEffect(() => {
-      startGame()
-   }, [])
+      allState.roundWinner === PLAYER_USER || allState.roundWinner === DRAW_PLAYER ? usersTurn(cards) : computersTurn(cards);
```

Comment:
Can't help but think we should join this `if` and this `ternary`, something like [null, PLAYER_USER, DRAW_PLAYER].includes(allState.roundWinner) ? usersTurn(cards) : computersTurn(cards)